editor/src/input/input_preprocessor.rs

```diff
@@ -77,15 +92,24 @@ impl MessageHandler<InputPreprocessorMessage, ()> for InputPreprocessor {
 				self.keyboard.unset(key as usize);
 				responses.push_back(InputMapperMessage::KeyUp(key).into());
 			}
-			InputPreprocessorMessage::ViewportResize(size) => {
-				responses.push_back(
-					graphene::Operation::TransformLayer {
-						path: vec![],
-						transform: glam::DAffine2::from_translation((size - self.viewport_size) / 2.).to_cols_array(),
-					}
-					.into(),
-				);
-				self.viewport_size = size;
+			InputPreprocessorMessage::BoundsOfViewports(bounds_of_viewports) => {
+				for bounds in bounds_of_viewports {
```

Comment:
Putting this in a for loop is just confusing

Reply:
I would prefer if it would at least panic if anything other than one viewport was supplied because this will otherwise lead to bugs in the future

Reply:
Why would iterating through a list in a for loop be confusing?

Reply:
because you set overwrite the value in each iteration which is unexpected